crates/store/re_protos/proto/rerun/v1alpha1/jobs.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+
+package rerun.redap_jobs.v1alpha1;
+
+import "rerun/v1alpha1/common.proto";
+
+// JobsService is the service for submitting and querying persistend redap tasks.
+service JobsService {
+  // Submit new tasks
+  rpc SubmitTasks(SubmitTasksRequest) returns (SubmitTasksResponse);
+  // Query the status of submitted tasks
+  rpc Query(QueryRequest) returns (QueryResponse);
+}
+
+message SubmitTasksRequest {
+  repeated Task tasks = 1;
+}
+
+message SubmitTasksResponse {}
+
+// A task is the unit of work in a tasks.
+message Task {
+  // Unique identifier for the task
+  string task_id = 1;
+  // Type of the task
+  string task_type = 2;
+  // Task-type dependant data necessary de-serialize the task
+  bytes task_data = 3;
+}
+
+// QueryRequest is the request message for querying tasks status
+message QueryRequest {
+  oneof query {
+    // Query all tasks in the system
+    QueryAllTasksRequest query_all_tasks = 1;
+    // Filter tasks by id
+    QueryFilterTasksRequest query_filter_tasks = 2;
+  }
+}
+
+message QueryAllTasksRequest {}
+message QueryFilterTasksRequest {
+  repeated string task_ids = 1;
+}
```

Comment:
`oneof` can create a lot of usability issues down the line -- my advice is to avoid it when it's not absolutely necessary.

Dropping it will also remove a bunch of typing layers in this instance:
```suggestion
// QueryRequest is the request message for querying tasks status
message QueryRequest {
  // If set, unconditionally fetch all tasks, always.
  bool fetch_all = 1;

  repeated string task_ids = 2;
}
```

At which point I would suggest to simply do this:
```suggestion
// QueryRequest is the request message for querying tasks status
message QueryRequest {
  // Empty queries for all tasks, if the server allows it, otherwise returns an error.
  repeated string task_ids = 1;
}
```